### PR TITLE
api: add Gang api

### PR DESCRIPTION
**Pull request by Wenshiqi222**
### Ⅰ. Describe what this PR does
add some consts for gang scheduling

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/extension/scheduling.go [hunk 1]
```diff
@@ -33,6 +34,31 @@ const (
...
+	GangNameAnnotation     = GangAnnotationPrefix + "/name"
+	GangMinNumAnnotation   = GangAnnotationPrefix + "/min-available"
+	GangWaitTimeAnnotation = GangAnnotationPrefix + "/waiting-time"
+	GangTotalNumAnnotation = GangAnnotationPrefix + "/total-number"
+	GangModeAnnotation     = GangAnnotationPrefix + "/gang-mode"
+	GangGroupsAnnotation   = GangAnnotationPrefix + "/groups"
+	GangTimeOutAnnotation  = GangAnnotationPrefix + "/timeout"
+
+	//Permit internal status
+	GangNotFoundInCache Status = "Gang not found in cache"
```
Comment (eahydra) [inline apis/extension/scheduling.go:57]: move theses status into GangPlugin.

File: apis/extension/scheduling.go [hunk 2]
```diff
@@ -33,6 +34,31 @@ const (
...
 )
 
+type Status string
+
+//Gang scheduling consts
+const (
+	GangAnnotationPrefix = "gang.scheduling.koordinator.sh"
+	DefaultGangWaitTime  = 2 * time.Minute
+
+	StrictMode    = "StrictMode"
```
Comment (eahydra) [inline apis/extension/scheduling.go:44]: Add GangMode prefix, such as `GangModeStrict`, `GangModeNonStrict`

File: apis/extension/scheduling.go [hunk 3]
```diff
@@ -33,6 +34,31 @@ const (
 	AnnotationReservationAllocated = SchedulingDomainPrefix + "/reservation-allocated"
 )
 
+type Status string
+
+//Gang scheduling consts
+const (
+	GangAnnotationPrefix = "gang.scheduling.koordinator.sh"
+	DefaultGangWaitTime  = 2 * time.Minute
```
Comment (eahydra) [inline apis/extension/scheduling.go:42]: the default wait time can be defined in the apis/scheduling/config. User can configure the time via the scheduler config file.

File: apis/extension/scheduling.go [hunk 4]
```diff
@@ -33,6 +32,24 @@ const (
...
 
+//Gang scheduling
+const (
+	//Gang Mode
+	GangModePrefix = "GangMode"
+	StrictMode     = GangModePrefix + "Strict"
+	NonStrictMode  = GangModePrefix + "NonStrict"
+
+	// Gang's Annotation
+	AnnotationGangPrefix   = "gang.scheduling.koordinator.sh"
```
Comment (jasonliu747) [inline apis/extension/scheduling.go:43]: should we put this domain const under `apis/extension/constants.go`? WDYT @eahydra

File: apis/extension/scheduling.go [hunk 5]
```diff
@@ -33,6 +32,24 @@ const (
 	AnnotationReservationAllocated = SchedulingDomainPrefix + "/reservation-allocated"
 )
 
+//Gang scheduling
+const (
+	//Gang Mode
+	GangModePrefix = "GangMode"
+	StrictMode     = GangModePrefix + "Strict"
+	NonStrictMode  = GangModePrefix + "NonStrict"
```
Comment (jasonliu747) [inline apis/extension/scheduling.go:40]: 我认为 @eahydra 在这里 https://github.com/koordinator-sh/koordinator/pull/409#discussion_r929838523 想表达的意思应该是声明两个常数 
```
StrictMode     = "GangModeStrict"
NonStrictMode  = "GangModeNonStrict"
```
而不是再单独声明一个 `GangModePrefix`

Comment (Wenshiqi222) [reply apis/extension/scheduling.go:40]: > 我认为 @eahydra 在这里 [#409 (comment)](https://github.com/koordinator-sh/koordinator/pull/409#discussion_r929838523) 想表达的意思应该是直接声明两个常数。
> 
> ```
> StrictMode     = "GangModeStrict"
> NonStrictMode  = "GangModeNonStrict"
> ```
> 
> 而不是再单独声明一个 `GangModePrefix`。当然，你这样也是可以的～

好的

File: apis/extension/scheduling.go [hunk 6]
```diff
@@ -33,6 +32,24 @@ const (
...
+	// Gang's Annotation
+	AnnotationGangPrefix   = "gang.scheduling.koordinator.sh"
+	AnnotationGangName     = AnnotationGangPrefix + "/name"
+	AnnotationGangMinNum   = AnnotationGangPrefix + "/min-available"
+	AnnotationGangWaitTime = AnnotationGangPrefix + "/waiting-time"
+	AnnotationGangTotalNum = AnnotationGangPrefix + "/total-number"
+	AnnotationGangMode     = AnnotationGangPrefix + "/gang-mode"
+	AnnotationGangGroups   = AnnotationGangPrefix + "/groups"
+	AnnotationGangTimeOut  = AnnotationGangPrefix + "/timeout"
+)
```
Comment (hormes) [inline apis/extension/scheduling.go:51]: PodGroup 上的 annotation 呢？

Comment (Wenshiqi222) [reply apis/extension/scheduling.go:51]: > PodGroup 上的 annotation 呢？

podGroup annotation 也用这个

File: apis/extension/scheduling.go (hunk 4 above)
Comment (eahydra) [reply apis/extension/scheduling.go:43]: Hmm... That's fine too..

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**